helpers.cmake

```diff
@@ -1,32 +1,41 @@
-function(include_src)
-    include_directories(src)
-    add_subdirectory(src/broker)
-    add_subdirectory(src/esp)
-    add_subdirectory(src/freeRtos)
-    add_subdirectory(src/gpio)
-    add_subdirectory(src/i2c)
-    add_subdirectory(src/network)
-    add_subdirectory(src/uart)
-    add_subdirectory(src/sht3x)
-    add_subdirectory(src/adxl345b)
-    add_subdirectory(src/pac193x)
-    add_subdirectory(src/spi)
-    add_subdirectory(src/fpgaConfigurationHandler)
-    add_subdirectory(src/middleware)
-    add_subdirectory(src/flash)
-    add_subdirectory(src/enV5HwController)
-    add_subdirectory(src/atCommands)
-    add_subdirectory(src/http)
-    add_subdirectory(src/sleep)
+function(add_cexception)
+    add_library(CException INTERFACE)
+    target_sources(CException INTERFACE ${CMAKE_CURRENT_FUNCTION_LIST_DIR}/extern/cexception/lib/CException.c)
+    target_include_directories(CException INTERFACE ${CMAKE_CURRENT_FUNCTION_LIST_DIR}/extern/cexception/lib)
 endfunction()
 
+function(add_unity)
+    add_subdirectory(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/extern/unity)
+endfunction()
+
+function(add_rp2040_hal)
+    add_subdirectory(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/src/hal)
+endfunction()
+
+function(add_basic_functionality)
+    add_cexception()
+    add_subdirectory(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/extern/elastic-ai.runtime.c)
+    add_subdirectory(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/src/common)
+    add_subdirectory(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/src/rtos)
+    add_subdirectory(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/src/network)
+    add_subdirectory(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/src/flash)
+    add_subdirectory(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/src/fpga)
+endfunction()
+
+function(add_sensor_libraries)
+    add_subdirectory(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/src/sensor)
+endfunction()
+
+
+
 function(make_to_output_file target)
```

Comment:
can you explain to me what this function does?

Reply:
I assume you refer to the `make_to_output_file` function. This function is there to set the correct values for hardware targets that run on the enV5 itself and copy the *.uf2 file to the `out` directory.
This can be done by passing the target to this function, or if we don't have this function by manually do all this for every target.

Reply:
so could we say it is a function you use to create an env5 executable file?

Reply:
can you change the name accordingly

Reply:
Renamed it to `create_enV5_executable`. Hope thats okay.